site/content/docs/06-server-side-rendering.md

```diff
@@ -2,19 +2,29 @@
 title: Server-side rendering
 ---
 
-Sapper, by default, renders server-side first (SSR), and then re-mounts any dynamic elements on the client. Svelte provides [excellent support for this](https://svelte.dev/docs#Server-side_component_API). This has benefits in performance and search engine indexing, among others, but comes with its own set of complexities.
+Sapper will render any component first on the server side and send it to the client as HTML. It will then run the component again on the client side to allow it to update based on dynamic data. This means you need to ensure that components can run both on the client and server side.
 
-### Making a component SSR compatible
+If, for example, your components try to access the global variables `document` or `window`, this will result in an error when the page is pre-rendered on the server side.
 
-Sapper works well with most third-party libraries you are likely to come across. However, sometimes, a third-party library comes bundled in a way which allows it to work with multiple different module loaders. Sometimes, this code creates a dependency on `window`, such as checking for the existence of `window.global` might do.
+If you need access to these variables, you can run code exclusively on the client side by wrapping it in
 
-Since there is no `window` in a server-side environment like Sapper's, the action of simply importing such a module can cause the import to fail, and terminate the Sapper's server with an error such as:
-
-```bash
-ReferenceError: window is not defined
+```js
+if (typeof window !== 'undefined') {
```

Comment:
What about `process.browser`? It's provided in both rollup and webpack default template, and I seem to see it being recommended more than checking `typeof window`.

Reply:
I think there's a bit of an ideological question here of how much we want to separate `sapper` from `sapper-template`. `process.browser` isn't something that's built in to Sapper - it's just something that happens to be replaced at bundle time by the default template. When it's available, `process.browser` _is_ the better thing to use because the replacement happens early enough that unused code can be removed.